packages/stencil-library/src/components/icon/icon.tsx

```diff
@@ -0,0 +1,44 @@
+import { Component, getAssetPath, h, Prop } from "@stencil/core";
+
+@Component({
+    tag: 'njwds-icon'
+})
+export class Icon {
+    @Prop() icon: string
+    @Prop() size: "3" | "4"| "scale" = "3"
+    @Prop() decorative: boolean = false;
+    @Prop() iconTitle?: string
+
+    componentWillLoad() {
+        if (!this.decorative && this.iconTitle === undefined) {
+            throw Error('<njwds-icon>: missing the "iconTitle" prop. The "iconTitle" prop is required unless the "decorative" prop is set to true.')
+        }
+    }
+
+    render() {
+        const spriteSrc = getAssetPath("img/sprite.svg");
+        const iconSrc = `${spriteSrc}#${this.icon}`
+
+        const sizeToClass = {
+            "scale": "",
+            "3": "usa-icon--size-3",
+            "4": "usa-icon--size-4"
+        }
```

Comment:
[Pebble] This could be a constant outside of the component scope to make it clearer that it's not dependent on the internals of the function or class